src/modules/historical.ts

```diff
@@ -70,10 +70,18 @@ export default function historical(
           const value = queryOptions[fieldName];
           if (value instanceof Date)
             queryOptions[fieldName] = Math.floor(value.getTime() / 1000);
-          else if (typeof value === "string")
-            queryOptions[fieldName] = Math.floor(
-              new Date(value as string).getTime() / 1000
-            );
+          else if (typeof value === "string") {
+            const timestamp = new Date(value as string).getTime();
+
+            if (isNaN(timestamp) || timestamp <= 0)
```

Comment:
`timestamp <= 0` is actually ok :)  because 0 is the unix epoch of Jan 1st, 1970 GMT.  So we use negative numbers for dates before that (try e.g. `new Date(-3_600_000) // -1hr` and see what pops up).  I don't know how far back Yahoo's data goes, but let's leave it up to the user to play with that :D  So let's keep just the `isNaN` check,

Reply:
You're right, that makes total sense. I updated the commit.